CONTRIBUTING.md

```diff
@@ -12,6 +12,29 @@ Origin (DCO). This document was created by the Linux Kernel community and is a
 simple statement that you, as a contributor, have the legal right to make the
 contribution. See the [DCO](DCO) file for details.
 
+## Expectations around PRs
+
+A branch submitted in a PR should consist only of a strict chain of
+commits built on a commit that was in the `main` branch --- no merges
+into the chain.  Rebase if necessary.  Do not make the chain
+unnecessarily long; squash together commits that have no useful
+distinction going forward.
+
+Unlike the automation for Kubernetes, our CI automation does not
```

Comment:
I suggest to move this section to the end. It is related only to people in OWNERS and is not relevant to most people. I think the next section about the LGTM requirements should come first.
Maybe even put it with some prefix of "Note to OWNERS ...."  or something